napari_arboretum/visualisation/vispy_plotter.py

```diff
@@ -0,0 +1,132 @@
+from typing import Optional
+
+import numpy as np
+from qtpy.QtWidgets import QWidget
+from vispy import scene
+
+from ..tree import Annotation, Edge
+from .base_plotter import TreePlotterQWidgetBase
+
+__all__ = ["VisPyPlotter"]
+
+
+class VisPyPlotter(TreePlotterQWidgetBase):
+    """
+    Tree plotter using pyqtgraph as the plotting backend.
+
+    Attributes
+    ----------
+    canvas : vispy.scene.SceneCanvas
+        Main plotting canvas
+    tree : TreeVisual
+        The tree.
+    """
+
+    def __init__(self):
+        """
+        Setup the plot canvas..
+        """
+        self.canvas = scene.SceneCanvas(keys=None, size=(300, 1200))
+        self.view = self.canvas.central_widget.add_view()
+        self.view.camera = scene.PanZoomCamera()
+        self.tree = TreeVisual(parent=None)
+        self.view.add(self.tree)
+
+    def get_qwidget(self) -> QWidget:
+        return self.canvas.native
+
+    def clear(self) -> None:
+        self.tree.clear()
+
+    def autoscale_view(self) -> None:
+        """Scale the canvas so all branches are in view."""
+        xs = np.concatenate([track.pos[:, 0] for id, track in self.tree.tracks.items()])
+        ys = np.concatenate([track.pos[:, 1] for id, track in self.tree.tracks.items()])
```

Comment:
I suspect these would work as generators? As in, you may be able to remove the `[]`